skrub/tests/test_datetime_encoder.py

```diff
@@ -1,494 +1,461 @@
-    }
-    enc.fit(X)
-    assert enc.features_per_column_ == expected_features_per_column_
-
-
-def test_extract_until_none() -> None:
-    X = get_dirty_datetime_array()
-    enc = DatetimeEncoder(extract_until=None)
-    expected_features_per_column_ = {
-        # all features after seconds are constant
-        # we want total_time if we have not extracted all non-constant features
-        0: ["total_time"],
-        1: ["total_time"],
-        2: ["total_time"],
-    }
-    enc.fit(X)
-    assert enc.features_per_column_ == expected_features_per_column_
+def test_to_datetime_incorrect_skip(X):
+    assert_array_equal(to_datetime(X), X)
 
-    # check get_names_out
-    expected_feature_names = [
-        "0_total_time",
-        "1_total_time",
-        "2_total_time",
-    ]
-    assert enc.get_feature_names_out() == expected_feature_names
 
-    # check with constant datetimes
-    X = get_constant_date_array()
-    enc = DatetimeEncoder(extract_until=None)
-    assert enc.fit_transform(X).shape[1] == 0
+def test_to_datetime_type_error():
+    # 3d tensor
+    X = [[["2021-01-01"]]]
+    with pytest.raises(TypeError):
+        to_datetime(X)
 
 
-def test_check_fitted_datetime_encoder() -> None:
-    """Test that calling transform before fit raises an error"""
-    X = get_datetime_array()[:, 0].reshape(-1, 1)
-    enc = DatetimeEncoder(add_day_of_the_week=True)
-    with pytest.raises(NotFittedError):
-        enc.transform(X)
+def test_to_datetime_invalid_params():
+    with pytest.raises(ValueError, match=r"(?=.*errors options)"):
+        to_datetime(2020, errors="skip")
 
-    # Check that it works after fit
-    enc.fit(X)
-    enc.transform(X)
+    with pytest.raises(ValueError, match=r"(?=.*not a parameter of skrub)"):
+        to_datetime(2020, unit="second")
+
+
+@pytest.mark.skipif(
+    not _is_pandas_format_mixed_available(),
+    reason=MSG_MIN_PANDAS_SKIP,
+)
+def test_to_datetime_format_param():
+    X_col = ["2021-01-01", "2021/01/01"]
+
+    # without format (default)
+    out = to_datetime(X_col)
+    expected_out = np.array(["2021-01-01", "NaT"], dtype="datetime64[ns]")
+    assert_array_equal(out, expected_out)
+
+    # with format
+    out = to_datetime(X_col, format="%Y/%m/%d")
+    expected_out = np.array(["NaT", "2021-01-01"], dtype="datetime64[ns]")
+    assert_array_equal(out, expected_out)
+
+
+def test_mixed_datetime_format():
+    df = get_mixed_datetime_format()
+
+    df_dt = to_datetime(df)
+    expected_df_dt = pd.DataFrame(
+        dict(
+            a=[
+                pd.Timestamp("2022-10-15"),
+                pd.Timestamp("2021-12-25"),
+                pd.Timestamp("2020-05-18"),
+                pd.Timestamp("2019-10-15 12:00:00"),
+            ]
+        )
+    )
+    assert_frame_equal(df_dt, expected_df_dt)
+
+    series_dt = to_datetime(df["a"])
+    expected_series_dt = expected_df_dt["a"]
+    assert_array_equal(series_dt, expected_series_dt)
+
+
+def test_mix_of_unambiguous():
```

Comment:
I think that the fact that this is a no-op is because YDM formats don't exist, not because of the mix of unambiguous.
For instance:
```
to_datetime(["10/15/2021", "13/01/2021"])
```
returns
```
array(['2021-10-15T00:00:00.000000000',                           'NaT'],
      dtype='datetime64[ns]')
```
for pandas > 2 (mixed otherwise).

Reply:
You're right, this is not what we want to test